src/python_testing/TC_EWATERHTR_2_1.py

```diff
@@ -0,0 +1,120 @@
+
+    def pics_TC_EWATERHTR_2_1(self):
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        return ["EWATERHTR.S"]
+
+    def steps_TC_EWATERHTR_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep("1", "Commissioning, already done",
+                     is_commissioning=True),
+            TestStep("2", "TH reads HeaterTypes attribute.",
+                     "DUT as Server replies with a WaterHeaterTypeBitmap (enum8) greater than 0x00 (at least one type supported), and less than 0x20 (no undefined types supported)."),
+            TestStep("3", "TH reads HeatDemand attribute.",
+                     "DUT as Server replies with a WaterHeaterDemandBitmap (enum8)."),
+            TestStep("4", "TH reads TankVolume attribute.",
+                     "DUT as Server replies with a uint16 value."),
+            TestStep("5", "TH reads EstimatedHeatRequired attribute.",
+                     "DUT as Server replies with an energy-mWh value."),
+            TestStep("6", "TH reads TankPercentage attribute.",
+                     "DUT as Server replies with a percent value."),
+            TestStep("7", "TH reads BoostState attribute.",
+                     "DUT as Server replies with a BoostStateEnum (enum8) value."),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_1(self):
+
+        em_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.EM']
+        tp_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.TP']
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_greater(heaterTypes, 0,
+                               f"Unexpected HeaterTypes value - expected {heaterTypes} > 0")
+        asserts.assert_less_equal(heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kOther,
+                                  f"Unexpected HeaterTypes value - expected {heaterTypes} <= WaterHeaterTypeBitmap.kOther")
+
+        self.step("3")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_greater(heatDemand, 0,
+                               f"Unexpected HeatDemand value - expected {heatDemand} > 0")
+        asserts.assert_less_equal(heatDemand, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kOther,
+                                  f"Unexpected HeatDemand value - expected {heatDemand} <= WaterHeaterDemandBitmap.kOther")
+
+        self.step("4")
+        if em_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="TankVolume")
+            asserts.assert_greater_equal(value, 0, f"Unexpected TankVolume value - expected {value} >= 0")
```

Comment:
Tank Volume is a uint16_t, so this assert is not necessary.

Reply:
Good spot - the test plan calls for it to be validated as a uint16 - we need to check if the python code will automatically validate the type (I think it does) if so this line can be removed.